spot_rl_experiments/spot_rl/envs/gaze_env.py

```diff
@@ -3,103 +3,197 @@
+        dont_pick_up (bool): Whether to pick up the object or not
+        max_episode_steps (int): Max episode steps
+
+    Returns:
+        config (Config): Updated config object
+    """
+    # Update the config based on the input argument
+    config.DONT_PICK_UP = dont_pick_up
+
+    # Update max episode steps based on the input argument
+    config.MAX_EPISODE_STEPS = max_episode_steps
+    return config
+
 
-    return traj
+class GazeController:
+    """
+    GazeController is used to gaze at, and pick given objects.
 
+    Args:
+        config (Config): Config object
+        spot (Spot): Spot object
 
-def close_enough(pos1, pos2):
-    dist = np.linalg.norm(np.array([pos1[0] - pos2[0], pos1[1] - pos2[1]]))
-    theta = abs(wrap_heading(pos1[2] - pos2[2]))
-    return dist < 0.1 and theta < np.deg2rad(2)
+    How to use:
+        1. Create a GazeController object
+        2. Call execute() method with the target object list
+        3. Call shutdown() to stop the robot
+
+    Example:
+        config = construct_config(opts=[])
+        spot = Spot("spot_client_name")
+        gaze_target_list = ["apple", "banana"]
+        gaze_controller = GazeController(config, spot)
+        gaze_results = gaze_controller.execute(gaze_target_list)
+        gaze_controller.shutdown()
+    """
+
+    def __init__(self, config, spot):
+        self.config = config
+        self.spot = spot
+
+        # Setup
+        self.policy = GazePolicy(config.WEIGHTS.GAZE, device=config.DEVICE)
+        self.policy.reset()
+
+        self.gaze_env = SpotGazeEnv(config, spot)
+        self.gaze_env.power_robot()
+
+    def execute(self, target_object_list, take_user_input=False):
+        """
+        Gaze at the target object list and pick up the objects if specified in the config
+
+        CAUTION: The robot will drop the object after picking it, please use objects that are not fragile
+
+        Args:
+            target_object_list (list): List of target objects to gaze at
+            take_user_input (bool): Whether to take user input for the success of the gaze
+
+        Returns:
+            gaze_success_list (list): List of dictionaries containing the target object name, time taken and success
+        """
+        gaze_success_list = []
+        print(f"Target object list : {target_object_list}")
+        for target_object in target_object_list:
+            observations = self.gaze_env.reset(target_obj_name=target_object)
+            done = False
+            start_time = time.time()
+            self.gaze_env.say(f"Gaze at target object - {target_object}")
+            while not done:
+                action = self.policy.act(observations)
+                observations, _, done, _ = self.gaze_env.step(arm_action=action)
+            self.gaze_env.say("Gaze finished")
+
+            # Ask user for feedback about the success of the gaze and update the "success" flag accordingly
+            success_status_from_user_feedback = True
```

Comment:
This should be `True` irrespective.

The way we record "success" is : 
`gaze_attempted and success_status_from_user_feedback`

So if gaze was not attempted, `status` will record `False`.
If gaze was attempted but user says False, "success" will still be `False`.

Reply:
In the case where we wouldn't be asking user input, the default value is `True`.

This would be similar to having `success=condition and True`

Thus success will hold `True` only when `condition` is `True.
In our case, condition is `grasp_attempted`